gptel-transient.el

```diff
@@ -247,18 +247,23 @@ which see."
       (if (not (file-readable-p gptel-crowdsourced-prompts-file))
           (progn (message "No crowdsourced prompts available")
                  (call-interactively #'gptel-system-prompt))
-        (with-temp-buffer
-          (insert-file-contents gptel-crowdsourced-prompts-file)
-          (goto-char (point-min))
-          (forward-line 1)
-          (while (not (eobp))
-            (when-let ((act (read (current-buffer))))
-              (forward-char)
-              (save-excursion
-                (while (re-search-forward "\"\"" (line-end-position) t)
-                  (replace-match "\\\\\"")))
-              (when-let ((prompt (read (current-buffer))))
-                (puthash act prompt gptel--crowdsourced-prompts)))
+	(with-temp-buffer
+	  (insert-file-contents gptel-crowdsourced-prompts-file)
+	  (goto-char (point-min))
+	  (forward-line 1)
+	  (while (not (eobp))
+            (let ((line (thing-at-point 'line t)))
+              (when line
+		;; Use a regex to split the line while respecting quoted strings
+		(let ((fields (split-string line ",\"?")))
```

Comment:
what happens if there's `,"` inside the prompt?

Reply:
You can do this by parsing the buffer instead of splitting the line into a string.  `search-forward` to the next unescaped `,` and collect the text from the start of search until here.